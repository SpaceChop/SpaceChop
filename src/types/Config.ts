import * as t from 'runtypes';
import validateParams from '../lib/validateParams';
import Paths from './Paths';
import Presets from './Presets';
import Sources from './Sources';
import Storage from './Storage';

const Config = t.Record({
  paths: Paths,
  sources: Sources,
  presets: Presets,
}).And(t.Partial({
  storage: Storage,
  disableChunkedEncoding: t.Boolean,
})).withConstraint(validateParams);

export type Config = t.Static<typeof Config>;
export default Config;
