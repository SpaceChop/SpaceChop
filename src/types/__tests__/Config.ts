import Config from '../Config';

describe('validate Config', () => {
  it('should fail invalid config', () => {
    const config = {};
    expect(() => Config.check(config)).toThrowErrorMatchingSnapshot();
  });
});
