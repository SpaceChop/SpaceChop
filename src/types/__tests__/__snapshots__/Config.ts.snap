// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Config should fail invalid config 1`] = `"Expected array, but was undefined"`;
